lmp/static_deltas.py

```diff
@@ -49,6 +52,29 @@ def save_delta_stats(delta_stats: dict, out_dir: str):
             f.write(s["canonical-json"])
 
 
+def upload_delta_stats(factory: str, delta_stats: dict, tok_secret_name: str):
+    ostreehub_uri = f"https://api.foundries.io/ota/ostreehub/{factory}/v2/repos/lmp/delta-stats"
+    for to_sha, s in delta_stats.items():
+        status(f"Uploading delta stats for {to_sha}...")
+        r = requests.put(ostreehub_uri,
+                         headers={
+                             "osf-token": secret(tok_secret_name),
+                             "content-type": "application/json",
+                             "content-digest": f"sha-256=:{s['sha256']}:"
+                         },
+                         data=s["canonical-json"])
+        if not r.ok:
+            raise requests.exceptions.HTTPError("Failed to upload delta stats to ostreehub; "
+                                                f"{r.status_code}, err: {r.text}")
+        # make sure the `sha` and `size` received from ostreehub matches with the original one
+        if r.json()["sha256"] != s["sha256"]:
+            raise Exception("invalid content hash is received from ostreehub; "
+                            f"expected: {s['sha256']}, received: {r.json()['sha256']}")
+        if int(r.json()["size"]) != len(s["canonical-json"]):
```

Comment:
should we convert size to int? ie if its not an int we should probably fail just do to a bad assertion in our data types.

Reply:
Won't this `if int(r.json()["size"])` raise an exception if `r.json()["size"]` is not an integer before making comparison.

Actually, I realize now, that I should have made ostreehub return an integer as a value of the "size" field. I don't know why I made it a string in the first place. I will update ostreehub at first, and then this code, so  `r.json()["size"]` becomes an integer.